### PR TITLE
Fix: Modifying settings doesn't apply immediately

### PR DESCRIPTION
**Pull request by allen-li1231**
### Related Issue
fixed on extension side rjmacarthy/twinny#6

### CODE REVIEW

File: src/completion.ts [hunk 1]
```diff
@@ -147,4 +149,15 @@ export class CompletionProvider implements InlineCompletionItemProvider {
...
+
+  public updateConfig() {
+    this._config = workspace.getConfiguration('twinny');
+    this._debounceWait = this._config.get('debounceWait') as number;
+    this._contextLength = this._config.get('contextLength') as number;
+    this._serverPath = this._config.get('server');
+    this._engine = this._config.get('engine');
+
+    this._basePath = `${this._serverPath}/${this._engine}`;
+    this._openai = new OpenAIApi(this._openaiConfig, this._basePath);
```
Comment (rjmacarthy) [inline src/completion.ts:161]: Could we move this out of `updateConfig`

File: src/completion.ts [hunk 2]
```diff
@@ -147,4 +149,15 @@ export class CompletionProvider implements InlineCompletionItemProvider {
...
   }
+
+  public updateConfig() {
+    this._config = workspace.getConfiguration('twinny');
+    this._debounceWait = this._config.get('debounceWait') as number;
+    this._contextLength = this._config.get('contextLength') as number;
+    this._serverPath = this._config.get('server');
+    this._engine = this._config.get('engine');
+
+    this._basePath = `${this._serverPath}/${this._engine}`;
```
Comment (rjmacarthy) [inline src/completion.ts:160]: Also maybe this?

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// A launch configuration that compiles the extension and then opens it inside a new window
```
Comment (rjmacarthy) [inline .vscode/launch.json:1]: Thanks for adding these, do we need the configurations, are they helpful?

File: src/extension.ts [hunk 4]
```diff
@@ -15,10 +15,12 @@ export function activate(context: ExtensionContext) {
   statusBar.text = '$(code)'
   statusBar.tooltip = 'twinny - Ready'
 
+  const completionProvider = new CompletionProvider(statusBar);
```
Comment (rjmacarthy) [inline src/extension.ts:18]: Please remove the semi colon.

File: src/completion.ts (hunk 1 above)
Comment (rjmacarthy) [inline src/completion.ts:161]: Please remove all semi colons.

**Comment by rjmacarthy** [comment]
LGTM, just some minor thing, thank you for your contribution!

File: src/completion.ts [hunk 5]
```diff
@@ -147,4 +149,15 @@ export class CompletionProvider implements InlineCompletionItemProvider {
       }) || []
     )
   }
+
+  public updateConfig() {
+    this._config = workspace.getConfiguration('twinny');
```
Comment (rjmacarthy) [inline src/completion.ts:154]: I think we can move this out too?

File: .vscode/launch.json (hunk 3 above)
Comment (allen-li1231) [reply .vscode/launch.json:1]: So what it defines is to launch a `npm-watch` task which subsequently looks for  `watch` config (which is `webpack --watch`) in your package.json when debugging task is launched.
What it does is to compile the source code whenever it is changed so developers won't have to restart debugging task everytime they make any changes to code. This file is supposed to be auto-generated when initializing a new extension project using vsce. I personally recommend adding these files to make debugging easier.

**Review by allen-li1231 — COMMENTED** [review]
Removed semi-colons as suggested

**Review by rjmacarthy — APPROVED** [review]

**Pull request merged**